File: Python/token.py
import http.client
import gzip
import json
from apikey import APIKEY

def get_token():
    # URL for token
    conn = http.client.HTTPSConnection("iam.cloud.ibm.com")

    # Payload for retrieving token. Note: An API key will need to be generated and replaced here
    payload = f'grant_type=urn%3Aibm%3Aparams%3Aoauth%3Agrant-type%3Aapikey&apikey={APIKEY}&response_type=cloud_iam'

    # Required headers
    headers = {
        'Content-Type': 'application/x-www-form-urlencoded',
        'Accept': 'application/json',
        'Cache-Control': 'no-cache'
    }

    try:
        # Connect to endpoint for retrieving a token
        conn.request("POST", "/identity/token", payload, headers)

        # Get and read response data
        res = conn.getresponse().read()
        data = res.decode("utf-8")

        # Format response in JSON
        json_res = json.loads(data)

        # Concatenate token type and token value
        return json_res['token_type'] + ' ' + json_res['access_token']

    # If an error happens while retrieving token
    except Exception as error:
        print(f"Error getting token. {error}")
        raise

File: Go/src/core/token.go
package core

import (
	"encoding/json"
	"io/ioutil"
	"log"
	"net/http"
	"strings"
)

// Token - struct for response type for token api
type Token struct {
	AccessToken  string `json:"access_token"`
	RefreshToken string `json:"refresh_token"`
	TokenType    string `json:"token_type"`
	ExpiresIn    int    `json:"expires_in"`
	Expiration   int    `json:"expiration"`
	Scope        string `json:"scope"`
}

// RetrieveToken - func to retrieve token
func RetrieveToken(apikey string) {
	// Create payload
	payloadSlice := []string{"grant_type=urn%3Aibm%3Aparams%3Aoauth%3Agrant-type%3Aapikey&response_type=cloud_iam&apikey=", apikey}
	payload := strings.NewReader(strings.Join(payloadSlice, ""))

	// Create a new request given a method, URL, and optional body.
	req, err := http.NewRequest("POST", IAM_endpoint, payload)
	if err != nil {
		log.Fatal(err)
	}

	// Adding headers to the request
	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
	req.Header.Add("Accept", "application/json")

	// Requesting server
	res, err := http.DefaultClient.Do(req)
	if err != nil {
		log.Fatal(err)
	}
	defer res.Body.Close()

	// Reading response
	body, err := ioutil.ReadAll(res.Body)
	if err != nil {
		log.Fatal(err)
	}

	// body is []byte format
	// parse the JSON-encoded body and stores the result in the struct object for the res
	var token Token
	json.Unmarshal([]byte(body), &token)
	// saving the token
	Iam_token = token.TokenType + " " + token.AccessToken
}
